snuba/datasets/events_processor_base.py

```diff
@@ -157,43 +154,26 @@ def process_message(
         * processed_message: dict representing the processed column -> value(s)
         Returns `None` if the event is too old to be written.
         """
-        action_type = None
+        version = message[0]
+        if version != 2:
+            raise InvalidMessageVersion(f"Unsupported message version: {version}")
 
-        if isinstance(message, dict):
-            # deprecated unwrapped event message == insert
-            action_type = ProcessorAction.INSERT
+        # version 2: (2, type, data, [state])
+        type_, event = message[1:3]
+        if type_ == "insert":
             try:
-                processed = self.process_insert(message, metadata)
+                return ProcessedMessage(
+                    ProcessorAction.INSERT, [self.process_insert(event, metadata)]
```

Comment:
How come we're passing something different to `process_insert` here?

Reply:
This is a bit hard to read in the diff view because of the removal of the unversioned block (it came from line 179 on the right hand side, not 166 as the alignment suggests here) but it didn't actually change for the versioned payload. In general, the unversioned payload is roughly the same structure as the versioned insertion payload (which is why we never really updated the tests to produce versioned payloads.)

Reply:
Cool, thanks